projects/Magnetosphere/Magnetosphere.cpp

```diff
@@ -169,12 +171,24 @@ namespace projects {
          RP::get(pop + "_Magnetosphere.taperOuterRadius", sP.taperOuterRadius);
          // Backward-compatibility: cfgs from before Sep 2021 setting pop_ionosphere.taperRadius will fail with the unknown option.
          // Some fail-safety checks
+         if(sP.taperInnerRadius < 0 || sP.taperOuterRadius < 0) {
+            if(myRank == MASTER_RANK) {
+               cerr << "Error: " << pop << "_Magnetosphere.taperInnerRadius and tapeOuterRadius should be >= 0! Aborting." << endl;
+            }
+            abort();
+         }
          if(sP.taperInnerRadius > sP.taperOuterRadius) {
             if(myRank == MASTER_RANK) {
-               cerr << "Error: " << pop << "_Magnetosphere.taperInnerRadius should be <= taperOuterRadius" << endl;
+               cerr << "Error: " << pop << "_Magnetosphere.taperInnerRadius should be <= taperOuterRadius! Aborting." << endl;
             }
             abort();
          }
+         if(sP.taperInnerRadius == 0 && sP.taperOuterRadius > 0) {
```

Comment:
If taperInnerRadius==0 and taperOuterRadius < ionosphereRadius, this might end up pear-shaped?

Reply:
Hence the test at 174 for you. :)

Reply:
but that's earlier - this check might put us in that region after that check has occurred

Reply:
But if the 174 triggers we abort. I don't follow.

Reply:
Aaah, now. So one more check to ensure taperOuterRadius is !=0 and > ionosphereRadius?